### PR TITLE
Fix death weapon not properly detonates

### PR DESCRIPTION
**Pull request by Thrifinesma**

### CODE REVIEW

**Comment by Lylaktech** [comment]
Tried it on all deathweapons in MO, no issues.

File: src/Misc/Hooks.BugFixes.cpp [hunk 1]
```diff
@@ -101,7 +102,7 @@ DEFINE_HOOK(0x702299, TechnoClass_ReceiveDamage_DebrisMaximumsFix, 0xA)
 				amountToSpawn = Math::min(amountToSpawn, totalSpawnAmount);
 				totalSpawnAmount -= amountToSpawn;
 
-				for ( ; amountToSpawn > 0; --amountToSpawn)
+				for (; amountToSpawn > 0; --amountToSpawn)
```
Comment (Metadorius) [inline src/Misc/Hooks.BugFixes.cpp:105]:
```suggestion
				for ( ; amountToSpawn > 0; --amountToSpawn)
```

File: src/Misc/Hooks.BugFixes.cpp [hunk 2]
```diff
@@ -173,12 +174,27 @@ DEFINE_HOOK(0x44377E, BuildingClass_ActiveClickWith, 0x6)
...
+// Author: Uranusian
+DEFINE_HOOK(0x70D77F, TechnoClass_FireDeathWeapon_ProjectileFix, 0x8)
+{
+	GET(TechnoClass*, pThis, ESI);
+	GET(BulletClass*, pBullet, EBX);
+	GET(CoordStruct*, pCoord, EAX);
+
+	pBullet->Target = pThis;
+	pBullet->SetLocation(*pCoord);
+	pBullet->Fire(true);
```
Comment (Metadorius) [inline src/Misc/Hooks.BugFixes.cpp:197]: Let's rename it to correspond to IDB.

File: src/Misc/Hooks.BugFixes.cpp [hunk 3]
```diff
@@ -173,12 +174,27 @@ DEFINE_HOOK(0x44377E, BuildingClass_ActiveClickWith, 0x6)
...
+
+// bugfix: DeathWeapon not properly detonates
+// Author: Uranusian
+DEFINE_HOOK(0x70D77F, TechnoClass_FireDeathWeapon_ProjectileFix, 0x8)
+{
+	GET(TechnoClass*, pThis, ESI);
+	GET(BulletClass*, pBullet, EBX);
+	GET(CoordStruct*, pCoord, EAX);
+
+	pBullet->Target = pThis;
```
Comment (Metadorius) [inline src/Misc/Hooks.BugFixes.cpp:195]: Is this one actually needed? I suggest to check if CreateBullet assigns target or not.

**Review by Metadorius — CHANGES_REQUESTED** [review]
Good, add doc / credits additionally and we'll merge.

**Review by Metadorius — APPROVED** [review]

**Pull request merged**